### ISSUE TITLE
BUG: `PeriodIndex` cannot parse dates out of nanoseconds bounds

### ISSUE THREAD

**Issue by aulemahal** [pandas-dev/pandas#51004]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
pd.PeriodIndex(['2100-01-01 23:00', '2301-01-01 23:00'], freq='H')
```

### Issue Description

On some specific installations, this fails with `OutOfBoundsDatetime: Out of bounds nanosecond timestamp: 2301-01-01 23:00:00`

Full traceback:

<details>

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/_libs/tslibs/period.pyx:1485, in pandas._libs.tslibs.period._extract_ordinal()

AttributeError: 'str' object has no attribute 'ordinal'

During handling of the above exception, another exception occurred:

OutOfBoundsDatetime                       Traceback (most recent call last)
Cell In[4], line 1
----> 1 pd.PeriodIndex(['2100-01-01 23:00', '2301-01-01 23:00'], freq='H')

File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/core/indexes/period.py:273, in PeriodIndex.__new__(cls, data, ordinal, freq, dtype, copy, name, **fields)
    270         data = PeriodArray(ordinal, freq=freq)
    271     else:
    272         # don't pass copy here, since we copy later.
--> 273         data = period_array(data=data, freq=freq)
    275 if copy:
    276     data = data.copy()

File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/core/arrays/period.py:977, in period_array(data, freq, copy)
    973     return PeriodArray(ordinals, dtype=dtype)
    975 data = ensure_object(arrdata)
--> 977 return PeriodArray._from_sequence(data, dtype=dtype)

File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/core/arrays/period.py:274, in PeriodArray._from_sequence(cls, scalars, dtype, copy)
    271 periods = np.asarray(scalars, dtype=object)
    273 freq = freq or libperiod.extract_freq(periods)
--> 274 ordinals = libperiod.extract_ordinals(periods, freq)
    275 return cls(ordinals, freq=freq)

File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/_libs/tslibs/period.pyx:1459, in pandas._libs.tslibs.period.extract_ordinals()

File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/_libs/tslibs/period.pyx:1494, in pandas._libs.tslibs.period._extract_ordinal()

File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/_libs/tslibs/period.pyx:2579, in pandas._libs.tslibs.period.Period.__new__()

File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/_libs/tslibs/parsing.pyx:367, in pandas._libs.tslibs.parsing.parse_time_string()

File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/_libs/tslibs/parsing.pyx:416, in pandas._libs.tslibs.parsing.parse_datetime_string_with_reso()

File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/_libs/tslibs/timestamps.pyx:1698, in pandas._libs.tslibs.timestamps.Timestamp.__new__()

File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/_libs/tslibs/conversion.pyx:249, in pandas._libs.tslibs.conversion.convert_to_tsobject()

File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/_libs/tslibs/conversion.pyx:523, in pandas._libs.tslibs.conversion._convert_str_to_tsobject()

File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/_libs/tslibs/conversion.pyx:506, in pandas._libs.tslibs.conversion._convert_str_to_tsobject()

File /exec/pbourg/.conda/xscen/lib/python3.10/site-packages/pandas/_libs/tslibs/np_datetime.pyx:212, in pandas._libs.tslibs.np_datetime.check_dts_bounds()

OutOfBoundsDatetime: Out of bounds nanosecond timestamp: 2301-01-01 23:00:00

</details>

I have tried:
Pandas 1.5.3 installed from conda, python 3.10.8 : Fails
Pandas 1.5.3 installed from source, python 3.8 : Works
Pandas 1.5.2 installed from conda, python 3.10.8 : Works
Pandas's main installed from source, python 3.8 : Works
Pandas's main installed from source, python 3.10.8 : Works

It thus seems that conda's pandas 1.5.3 is at fault here.

I can try more configurations and add them to here.

### Expected Behavior

The date is parsed successfully.

### Installed Versions

In the environment where it fails:

<details>

INSTALLED VERSIONS
------------------
commit           : 2e218d10984e9919f0296931d92ea851c6a6faf5
python           : 3.10.8.final.0
python-bits      : 64
OS               : Linux
OS-release       : 3.10.0-1160.81.1.el7.x86_64
Version          : #1 SMP Fri Dec 16 17:29:43 UTC 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : fr_CA.UTF-8
LOCALE           : fr_CA.UTF-8

pandas           : 1.5.3
numpy            : 1.23.5
pytz             : 2022.7.1
dateutil         : 2.8.2
setuptools       : 58.5.3
pip              : 22.3.1
Cython           : None
pytest           : 7.2.1
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.8.0
pandas_datareader: None
bs4              : None
bottleneck       : 1.3.5
brotli           : 
fastparquet      : None
fsspec           : 2022.11.0
gcsfs            : None
matplotlib       : 3.6.2
numba            : 0.56.4
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 10.0.1
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.8.1
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : 2022.12.0
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None

</details>

**Comment by topper-123** [comment]
I've got this working on Pandas 2.0 without raising (macOS).

Can you try again on Pandas 2.0 on your machine? Datatime functionality got a major upgrade on pandas 2.0, so stands to reason that it works now.

Pandas 1.5.x will not get bug fixes, so if this is fixed in 2.0, the issue can be considered fixed.

**Comment by aulemahal** [comment]
Indeed, this seems fixed in pandas 2.

I'll be closing. Thanks!
